.github/workflows/build-windows.yml

```diff
@@ -0,0 +1,297 @@
+          Write-Host "BUILD_SOURCESDIRECTORY is '$env:BUILD_SOURCESDIRECTORY'"
+
+          #directory $env:BUILD_BUILDDIRECTORY was created in step 'tiledb env prep'
+          cd $env:BUILD_BUILDDIRECTORY
+
+          $VSCategory = "Enterprise" # alternate 'Community'
+          if ($env:TILEDB_GA_IMAGE_NAME -eq "windows-2016") {
+            #if (!(Test-Path "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin")) {
+            if (!(Test-Path "C:\Program Files (x86)\Microsoft Visual Studio\2017\${VSCategory}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin")) {
+              #Write-Host "ERROR***: Missing C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+              Write-Host "ERROR***: Missing C:\Program Files (x86)\Microsoft Visual Studio\2017\${VSCategory}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+              exit $LastExitCode
+            }
+            #$env:Path += ";C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+            $env:Path += ";C:\Program Files (x86)\Microsoft Visual Studio\2017\${VSCategory}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+          } elseif ($env:TILEDB_GA_IMAGE_NAME -eq "windows-2019") {
+            #if (!(Test-Path "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin")) {
+            if (!(Test-Path "C:\Program Files (x86)\Microsoft Visual Studio\2019\${VSCategory}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin")) {
+              #Write-Host "ERROR***: Missing C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+              Write-Host "ERROR***: Missing C:\Program Files (x86)\Microsoft Visual Studio\2019\${VSCategory}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+              exit $LastExitCode
+            }
+            #$env:Path += ";C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+            $env:Path += ";C:\Program Files (x86)\Microsoft Visual Studio\2019\${VSCategory}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+          } else {
+            Write-Host "Unknown image name: '$($env:TILEDB_GA_IMAGE_NAME)'"
+            exit $LastExitCode
+          }
+          
+          # allow double-checking path
+          cmd /c "echo $PATH"
+
+          # TODO DEBUG move this back in the TILEDB_S3 section
+          # currently we do not run S3 tests on Windows because tests time out (minio+azure slow?)
+          #& "$env:BUILD_SOURCESDIRECTORY\scripts\install-minio.ps1"
+
+          $bootstrapOptions = "-EnableVerbose -EnableStaticTileDB -EnableBuildDeps"
+          if ($env:TILEDB_S3 -eq "ON") {

```

Comment:
IMO, we could simplify this (and, eventually, our other scripts) significantly by just passing all of the bootstrap selections in the matrix config, rather than implementing parsing for all of the `TILEDB_*` options.

Reply:
(it would be great to see us minimize scripting inside of yaml whenever possible -- given the lack of local-testability of GHA -- hence this suggestion. But that's mostly a broader discussion)

Reply:
I've been tempted (that's what we do in tiledb-py for expedience), but if we are going to keep bootstrap.ps1 around, probably good to use it here so it is tested.

Reply:
set (in outer-most layer) TILEDB_BASE_BOOTSTRAP_OPTIONS then ref'd here,
TILEDB_BASE_BOOTSTRAP_OPTIONS should then be overridable in matrix if needed,
Does this satisfy what you had in mind?

Reply:
^^@ihnorton